BaseTools/Edk2ToolsBuild.py

```diff
@@ -134,8 +134,12 @@ def Go(self):
 
             # Actually build the tools.
             output_stream = edk2_logging.create_output_stream()
-            ret = RunCmd('nmake.exe', None,
-                         workingdir=shell_env.get_shell_var("EDK_TOOLS_PATH"))
+            if ((os.environ.get('CLANG_MAKEFILE_TYPE') == 'gnu') and (not self.tool_chain_tag.lower().startswith("vs"))):

```

Comment:
You bring up a good point. We should probably change this environment variable to <TOOL_CHAIN_TAG>_MAKEFILE_TYPE so that if you want to explicitly override the makefile behavior, you can go so for each tool chain.

So, for example, one could define environment variables like this:

```
VS2019_MAKEFILE_TYPE=nmake
GCC5_MAKEFILE_TYPE=gnu
CLANGPDB_MAKEFILE_TYPE=gnu
```

And so, on depending on how the user wants their build to run.